libcxx/include/__tree

```diff
@@ -945,8 +921,8 @@ public:
     return std::addressof(__end_node()->__left_);
   }
 
-  typedef __tree_iterator<value_type, __node_pointer, difference_type> iterator;
-  typedef __tree_const_iterator<value_type, __node_pointer, difference_type> const_iterator;
+  typedef __tree_iterator<_Tp, __node_pointer, difference_type> iterator;
```

Comment:
Since this is ABI sensitive and you're visibly tip-toeing around an ABI break here, we should have a test on that. Otherwise, a future refactoring could definitely break this without ever noticing.

For watchers, the issue is that `__tree_iterator` is ABI facing, and `_Tp` is sometimes `__value_type`, whereas `value_type` is an actual `pair<K const, V>`. Changing from `_Tp` to `value_type` here would change the mangling of the `iterator` typedef.